src/lexer/lexer.go

```diff
@@ -0,0 +1,482 @@
+package lexer
+
+import (
+	"fmt"
+	"github.com/gojinja/gojinja/src/errors"
+	"github.com/gojinja/gojinja/src/utils"
+	"github.com/gojinja/gojinja/src/utils/stack"
+	"github.com/hashicorp/golang-lru"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+var lexerCache *lru.Cache
+
+func init() {
+	var err error
+	lexerCache, err = lru.New(50)
+
+	if err != nil {
+		_, _ = fmt.Fprintln(os.Stderr, "%w", err)
+		os.Exit(1)
```

Comment:
A library should never exit(), but I guess it's ok here. It could probably even be changed to `panic`